File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        // sort(s.begin(),s.end());
        // sort(t.begin(),t.end());
        // if(s.size()!=t.size()) return false;
        // for(int i=0;i<s.size();i++) {
        //     if(s[i]!=t[i]) return false;
        // }return true; O(nlog n)
        if(s.size()!=t.size()) return false;
        int count[26] = {0};
        for(int i=0;i<s.size();i++) {
            count[s[i]-'a'] ++;
            count[t[i]-'a'] --;
        }
        for(int i=0;i<26;i++) {
            if(count[i]!=0) {
                return false;
            }
        }return true; //O(n)
    }
};

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length() != t.length()){
            return false;
        }
        
        int[] arr = new int[26];
        
        for(int i = 0; i< s.length(); i++){
            char c = s.charAt(i);
            arr[c-'a']++;
            char k = t.charAt(i);
            arr[k-'a']--;

        }

        for(int x : arr){
            if(x != 0){
                return false;
            }
        }
        return true;
    }
}